packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -112,18 +112,19 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
 
   async testTorrentingCapability(timeOut: number) {
     log('Testing torrenting capability...');
+    let torrentId;
     const gotAWire = new Promise(resolve => {
       super.add(mockTorrents[0].magnetURI, (torrent: Torrent) => {
-        torrent.once('wire', wire => {
-          this.remove(torrent.infoHash);
-          resolve(true);
-        });
+        torrentId = torrent.infoHash;
+        torrent.once('wire', wire => resolve(true));
       });
     });
     const timer = new Promise(function(resolve, reject) {
       setTimeout(resolve, timeOut);
     });
-    return Promise.race([gotAWire, timer]);
+    const raceResult = await Promise.race([gotAWire, timer]);
+    this.remove(torrentId);
```

Comment:
👍

Reply:
uh, I forgot I left that there. It avoids some inconsistent behaviour and some console errors that I noticed.